src/scripts/core/contour-utils.js

```diff
@@ -34,7 +36,16 @@
                 return generalHelpers.getValue(object, null, ctx);
             } else if (typeof object === 'object') {
                 return Object.keys(object).reduce(function (prev, key) {
-                    prev[key] = generalHelpers.materialize(object[key], ctx);
+                    var curKeyPath = curPath ? curPath + '.' + key : key;
+                    var notInSkipList =  skipList.indexOf(curKeyPath) === -1;
+                    var notSkipMatch = (skipMatch && skipMatch.test ? !skipMatch.test(curKeyPath) : true);
+                    var expectsParam = /(function)?\s?\(.+\)\s*({|=>)/.test((object[key] || '').toString());
```

Comment:
I think you can just do `object[key].length` to get how many parameters it takes.

Reply:
Why do we need to separate functions which specify params? For the ones which matter, like the formatter, can't we just blacklist them with the skip lists? Seems a bit scary to think `marker : { enable: ()=> true }` will work but `marker : { enable: (randomCopyPasteParameter)=> true }` won't.

Reply:
I was thinking the same thing... But otherwise, for regular callback functions like formatters, you have to remember to add the skip list which would not be very intuitive. We could leave this option out if you want.

Reply:
You mean "you" as in "Contour developer", not "end user" right? i.e., we'd already know what to blacklist and it'll only be a problem if we wanted to add  a new Contour options down the line which took in a function-with-param value.  

Or would you need to use this when you're building a custom visualization?

Reply:
I think this impacts two groups:

-  contour core developers, when adding new functionality. I guess this is less of a problem since you "we know what we are doing" :)
- Contour lib users. If you are developing a custom visualization that calls a function defined in your part of the config object, you'll need provide a default `skip` list that includes your function, but if the user of your visualization overrides the skip list, then the viz can break without much indication as of why. For example, let's say I create a custom viz like this:

```
const renderer = (data, layer, options) => {
  layer.selectAll('.translations').data(data);

  layer.enter()
    .append('div')
    .text(options.translate.getTranslation);
};

renderer.defaults = {
  translate: {
    getTranslation: (d) => `data is ${d.x}, ${d.y}`,
  },
  skip: ['translate.getTranslation']
}

Contour.export('translate', renderer);
```

in this case, if the user of the visualization passes a different skip list, `translate` viz would break.

Makes sense? maybe we can force to the functions to have a specific signature to prevent them from being materialized, but again consumers of custom viz would have to remember to follow the signature when passing a fn param.

In general my thinking was that the case were you pass a function that has a parameter by mistake is less probable than the case where you need to remember to add it to the skip list.

Another idea could be to have a "debug" mode in contour where you get debugging/console messages that tell you which functions got materialized.

Reply:
That makes sense - how about a whitelist of 'known materializable properties' instead of a blacklisted skip list then? This would mean there is no change in behavior for developing custom viz, unless we allow them to 'opt-in' by passing in an 'allow' flag (or maybe we just don't expose this to custom viz).

That said, you know a lot more about the different cases than I do so I'm fine going with your call either way.

Reply:
I thought about going with the whitelist route too, but then that would mean that mostly everything would be in the whitelist (as most props are fine to have functions) and if I create a custom viz, I need to add each property that _can_ be materializable to the list except from my little lonely `getTranslations` function... that seem an overkill too :)

Reply:
I meant whitelist specifically for custom visualizatons; if i'm defining a custom viz which takes in a boolean prop I wouldn't automatically expect that to be callable with a function
-- or does that distinction between external/internal visualizations not exist?